overlord/snapstate/handlers.go

```diff
@@ -1074,3 +1143,141 @@ func (m *SnapManager) doAliasV2(t *state.Task, _ *tomb.Tomb) error {
 	Set(st, snapName, snapst)
 	return nil
 }
+
+func (m *SnapManager) doDisableAliasesV2(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+	snapName := snapsup.Name()
+
+	oldAutoDisabled := snapst.AutoAliasesDisabled
+	oldAliases := snapst.Aliases
+	newAliases, _ := disableAliases(oldAliases)
+
+	if !snapst.AliasesPending {
+		if err := applyAliasesChange(st, snapName, oldAutoDisabled, oldAliases, true, newAliases, m.backend); err != nil {
+			return err
+		}
+	}
+
+	t.Set("old-auto-aliases-disabled", oldAutoDisabled)
+	snapst.AutoAliasesDisabled = true
+	t.Set("old-aliases-v2", oldAliases)
+	snapst.Aliases = newAliases
+	Set(st, snapName, snapst)
+	return nil
+}
+
+func (m *SnapManager) doUnaliasV2(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+	var alias string
+	err = t.Get("alias", &alias)
+	if err != nil {
+		return err
+	}
+	snapName := snapsup.Name()
+
+	autoDisabled := snapst.AutoAliasesDisabled
+	oldAliases := snapst.Aliases
+	newAliases, err := manualUnalias(oldAliases, alias)
+	if err != nil {
+		return err
+	}
+
+	if !snapst.AliasesPending {
+		if err := applyAliasesChange(st, snapName, autoDisabled, oldAliases, autoDisabled, newAliases, m.backend); err != nil {
+			return err
+		}
+	}
+
+	t.Set("old-aliases-v2", oldAliases)
+	snapst.Aliases = newAliases
+	Set(st, snapName, snapst)
+	return nil
+}
+
+type otherDisabledAliases struct {
+	Disabled bool              `json:"disabled,omitempty"`
+	Manuals  map[string]string `json:"manuals,omitempty"`
+}
+
+func (m *SnapManager) doPreferAliasesV2(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+	snapName := snapsup.Name()
+
+	autoDisabled := snapst.AutoAliasesDisabled
+	if !autoDisabled {
+		// already enabled, nothing to do
+		return nil
+	}
+
+	curAliases := snapst.Aliases
+	conflicts, err := checkAliasesConflicts(st, snapName, false, curAliases, nil)
+	if err != nil && conflicts == nil {
```

Comment:
tweaked, with the change the first return argument from checkAliasesConflict is a bit pointless, I suppose I could remove it but this branch is already big, it would be for a follow up